usr/bin/cylc

```diff
@@ -145,23 +145,33 @@ fi
 # Note: the code above is only needed if still using standalone Rose 2019
 # versions
 
+# CYLC_NAME construction:
+#                           ╔════════════════════════════╗
+#                           ║        CYLC_VERSION        ║
+#                           ╠══════════════════╦═════════╣
+#                           ║        set       ║ not set ║
+# ╔═══════════════╦═════════╬══════════════════╩═════════╣
+# ║ CYLC_ENV_NAME ║   set   ║      Use CYLC_ENV_NAME     ║
+# ║               ╠═════════╬══════════════════╦═════════╣
+# ║               ║ not set ║ Use CYLC_VERSION ║  "cylc" ║
+# ╚═══════════════╩═════════╩══════════════════╩═════════╝
 if [[ -z "${CYLC_HOME}" ]]; then
-    # Look for specified or default version under the root locations
-    if [[ -n "${CYLC_VERSION}" ]]; then
-        CYLC_NAME="cylc-$CYLC_VERSION"
-    else
+    if [[ -n "${CYLC_VERSION}" && -z "${CYLC_ENV_NAME}" ]]; then
+        CYLC_ENV_NAME="cylc-$CYLC_VERSION"
+    elif [[ -z "${CYLC_ENV_NAME}" && -z "${CYLC_VERSION}" ]]; then
         # Default version symlink
-        CYLC_NAME="cylc"
+        CYLC_ENV_NAME="cylc"
     fi
     for ROOT in "${CYLC_HOME_ROOT}" "${CYLC_HOME_ROOT_ALT}"; do
-        if [[ -d "${ROOT}/${CYLC_NAME}" ]]; then
-            CYLC_HOME="${ROOT}/${CYLC_NAME}"
+        if [[ -d "${ROOT}/${CYLC_ENV_NAME}" ]]; then
+            CYLC_HOME="${ROOT}/${CYLC_ENV_NAME}"
+            export CYLC_ENV_NAME
```

Comment:
I don't think we need this `export`?  I guess it would allow a tiny shortcut in the scripting above if `cylc` is invoked again in subshells, but it would be clearer to have subshells work out the path in exactly the same way as the parent.

Ping @oliver-sanders as it was evidently his suggestion?

Reply:
We may have multiple environments containing the same Cylc version which could lead to different results when activating via the `CYLC_VERSION` or `CYLC_ENV_NAME`.

Reply:
I don't think we even need to set it here. By this point, it is used to set CYLC_HOME, and that's what's used to activate the environment.

Reply:
I think the only case where we need to `export CYLC_ENV_NAME` is when we're using the default symlink version. We have to export it because cylc will always set `CYLC_VERSION` and we can no longer assume that the symlink version points to `cylc-$CYLC_VERSION`.